src/_guides/language/effective-dart/design.md

```diff
@@ -658,50 +696,121 @@ can.
+fields" in Dart, fields are "particularly fast getters".
+
+Even so, choosing a getter over a method sends an important signal to the caller
+and you should use a getter when that's the signal you want to send. The signal,
+roughly, is that the operation is "field-like". The operation, at least in
+principle, *could* be implemented using a field, as far as the caller knows.
+That implies:
+
+*   **The operation does not take any arguments and returns a result.**
+
+*   **The caller cares mostly about the result.** If you want the caller to
+    worry about *how* the operation produces its result more than they do the
+    result being produced, then give the operation a verb name that describes
+    the work and make it a method.
+
+    This does *not* mean the operation has to be particularly fast in order to
+    be a getter. `IterableBase.length` is `O(n)`, and that's OK. It's fine for a
+    getter to do significant calculation. But if it does a *surprising* amount
+    of work, you may want to draw their attention to that by making it a method
+    whose name is a verb describing what it does.
+
+    {:.bad-style}
+    {% prettify dart %}
+    connection.nextIncomingMessage; // Does network IO.
+    expression.normalForm; // Could be exponential to calculate.
+    {% endprettify %}
 
-  * **Does not take any arguments.**
-  * **Returns a value.**
-  * **Is side-effect free.** Invoking a getter shouldn't change any
-  externally-visible state (caching internally or lazy initialization is
-  OK). Invoking the same getter repeatedly should return the same value
-  unless the object is explicitly changed between calls.
+*   **The operation does not have user-visible side effects.** Accessing a real
+    field does not alter the object or any other state in the program. It
+    doesn't produce output, write files, etc. A getter should generally follow
+    that.
 
-{:.good-style}
-{% prettify dart %}
-rectangle.width
-collection.isEmpty
-button.canShow
-{% endprettify %}
+    The "user-visible" part is important. It's fine for getters to modify hidden
+    state or produce out of band side effects. Getters can lazily calculate and
+    store their result, write to a cache, log stuff, etc. As long as the caller
+    doesn't *care* about the side effect, it's probably fine.
 
-{:.bad-style}
+    {:.bad-style}
+    {% prettify dart %}
+    stdout.newline; // Produces output.
+    list.clear; // Modifies object.
+    {% endprettify %}
+
+*   **The operation is *idempotent*.** "Idempotent" is an odd word that, in this
+    context, basically means that calling the operation multiple times produces
+    the same result each time, unless some state is explicitly modified between
+    those calls. (Obviously, `list.length` produces different results if you add
+    an element to the list between calls.)
+
+    "Same" here does not mean a getter must literally produce an identical
```

Comment:
Same -> Same result
(I'd lost track of _what_ was the same by this paragraph.)

Reply:
Done.